texastribune/donations
# Portal + front-end Auth0 integration and basic Vue routing

#### What's this PR do?
+ Lots of file reorganization
+ Integrates log in/log out flow via Auth0
+ Adds Auth0 access token to all API requests
+ Adds basic Vue routing for future SPA

NOTE: It'll look hideous! No styles have been applied.

#### Why are we doing this? How does it help us?
Just UMPing along.

#### How should this be manually tested?
*You'll need a non-staff Auth0 account in our test tenant for this (and future UMP reviews)*

+ Ensure you have test values for `AUTH0_CLIENT_ID`, `AUTH0_CLIENT_SECRET` and `AUTH0_DOMAIN` in `env-docker`
+ Ensure you're logged out of any non-staff accounts in our Auth0 test tenant
+ `make`
+ `make restart` (wait for Webpack to finish)
+ Go to `/user-portal`. Confirm it redirects you to our universal log-in page.
+ Log in. Confirm you get redirected back to `/user-portal`, where the page will hang for a few (eventually it'll show a loading icon) while it checks your Auth0 session. After a few, you should see "this is home" and a few links.
+ Click those links. Confirm the text changes along with the URL, indicating the client-side routing is working.
+ Open up Chrome developer tools. Go to the network tab, then click "XHR" below it. Do a hard refresh, then look for a GET request to `content` (a temporary API call I dropped in for PR-review purposes). Click into it and look for "request headers." One should be `Authorization: Bearer <random string>`. That means our Auth0 access token is getting sent along with API calls.
+ Click "log out." You should get redirected to `/donate` afterward (that flow is up for debate). Go back to `/user-portal` and confirm you once again get prompted to log in.

#### How should this change be communicated to end users?
NA.

#### Are there any smells or added technical debt to note?
NA. Question for @x110dc: Right now, every API call will include the Auth0 access token in the Authorization header. Is there any security danger in that if, for example, we ever need to call a non-Texas Tribune API in this app? Can third parties do anything with that access token?

#### What are the relevant tickets?
https://3.basecamp.com/3098728/buckets/10357222/todos/1720701890

#### Have you done the following, if applicable:
* [ ] Added automated tests? *(later)*
* [ ] Tested manually on mobile? *(will do later when app is further along)*
* [ ] Checked for security implications? *(see question above)*
* [ ] Updated the documentation/wiki? *(later)*

#### TODOs / next steps:
Formalize Vue route-based data fetching.
